store/branchkv/iterator.go

```diff
@@ -0,0 +1,195 @@
+package branchkv
+
+import (
+	"slices"
+
+	"cosmossdk.io/store/v2"
+)
+
+var _ store.Iterator = (*iterator)(nil)
+
+// iterator walks over both the KVStore's changeset, i.e. dirty writes, and the
+// parent iterator, which can either be another KVStore or the SS backend, at the
+// same time.
+//
+// Note, writes that happen on the KVStore over an iterator will not affect the
+// iterator. This is because when an iterator is created, it takes a current
+// snapshot of the changeset.
+type iterator struct {
+	parentItr store.Iterator
+	start     []byte
+	end       []byte
+	key       []byte
+	value     []byte
+	keys      []string
+	values    []store.KVPair
+	reverse   bool
+	exhausted bool // exhausted reflects if the parent iterator is exhausted or not
+}
+
+func newIterator(parentItr store.Iterator, start, end []byte, cs map[string]store.KVPair, reverse bool) *iterator {
```

Comment:
how about creating the mapIterator for `cs` iteration? IMO, it would be more clear and reduce memory???

Reply:
You mean implement an `Iterate` method on the `Changeset` type?

Reply:
yes, changeset or map structure whatever

Reply:
Sure!